internal/service/domain/sensor/sensor_test.go

```diff
@@ -366,6 +367,88 @@ func TestSensorService_Delete(t *testing.T) {
 	})
 }
 
+func TestSensorService_RunStatusUpdater(t *testing.T) {
+	t.Run("should update stale sensor statuses periodically", func(t *testing.T) {
+		ctx := context.Background()
+		sensorRepo := storageMock.NewMockSensorRepository(t)
+		svc := SensorService{sensorRepo: sensorRepo}
+
+		staleSensor := &entities.Sensor{
+			ID:        "sensor-1",
+			UpdatedAt: time.Now().Add(-73 * time.Hour), // 73 hours ago
+		}
+		recentSensor := &entities.Sensor{
+			ID:        "sensor-2",
+			UpdatedAt: time.Now().Add(-1 * time.Hour), // 1 hour ago
+		}
+
+		sensorRepo.EXPECT().GetAll(mock.Anything).Return([]*entities.Sensor{staleSensor, recentSensor}, nil)
+		sensorRepo.EXPECT().Update(mock.Anything, staleSensor.ID, mock.Anything).Return(staleSensor, nil)
+
+		go func() {
+			svc.RunStatusUpdater(ctx, 10*time.Millisecond)
+		}()
+
+		time.Sleep(100 * time.Millisecond)
+
+		// No assertions needed since mock expectations handle it
```

Comment:
Why, I don't think so. Mock expectations will just return what you define here, which is true, but you need to validate that `svc.RunStatusUpdater` is doing the write thing.

Reply:
How can we do that if runStatusUpdater is only responsible for running the ticker in a loop?

Reply:
Do you have a better Idea?

Reply:
Where do you use `sensorRepo.AssertCalled` and `sensorRepo.AssertNotCalled`? I do not see it. Then this would be your validation and okay

Reply:
How are they not visible to you? Check this link: [internal/service/domain/sensor/sensor_test.go](https://github.com/green-ecolution/green-ecolution-backend/pull/262/files#diff-e7f09c8b46076b203bf75664c16e541c6f6a412fa390fbe4d1a3fccf30bde18e)